### PR TITLE
Enable setting color of text and images on unselected tabs

### PR DESCRIPTION
**Pull request by urbancvek**
Hi, 
This PR Solves this issue #3083.
This PR solves the problem of default color on TabBar being always grey. Which looks great if the barTintColor is unchanged. However if we set the barTintColor to something else (like blue in example) text and icons become quite unreadable.
simulator screen shot 27 apr 2016 21 58 40

Commit (c206417) - Enable setting color of unselected tabs
Solves this issue with a prop (unselectedTintColor) on TabBarIOS to which you just pass a color like you can for barTintColor and tintColor. 
This leaves us with a result that is on second picture. Notice the color of text on tabs.
simulator screen shot 27 apr 2016 21 59 06

Or change it to yellow for demonstrating purposes
simulator screen shot 27 apr 2016 21 59 13

Commit (f9ad80a) -	Enable rendering image of icon as original
Solves the issue of images passed as an icons being always rendered as a template (in grey color). If we want to have a colored image we can't do that. This is solved with a prop (renderAsOriginal) on TabBarIOS.Item.
This is the final version we can now achieve.
simulator screen shot 27 apr 2016 21 59 29

Or even something like this.
simulator screen shot 27 apr 2016 22 00 01

### CODE REVIEW

**Comment by urbancvek** [comment]
I would also be happy to write docs for this feature.

**Comment by facebook-github-bot-3** [comment]
By analyzing the blame information on this pull request, we identified @nicklockwood and @shtefanntz to be potential reviewers.

**Comment by facebook-github-bot-3** [comment]
Thank you for your pull request and welcome to our community. We require contributors to sign our Contributor License Agreement, and we don't seem to have you on file. In order for us to review and merge your code, please sign up at https://code.facebook.com/cla - and if you have received this in error or have any questions, please drop us a line at cla@fb.com. Thanks!

**Comment by facebook-github-bot-7** [comment]
@urbancvek updated the pull request.

**Comment by facebook-github-bot-9** [comment]
Thank you for signing our Contributor License Agreement. We can now accept your code for this (and any) Facebook open source project. Thanks!

**Comment by christopherdro** [comment]
@urbancvek Docs are pulled from the source so you can add a more descriptive message there.

**Comment by facebook-github-bot-6** [comment]
@urbancvek updated the pull request.

**Comment by urbancvek** [comment]
I've added an example to UIExplorer app so TabBarIOS Example now looks like this.
simulator screen shot 27 apr 2016 22 18 32

I've already written docs for API. :)

**Comment by christopherdro** [comment]
👍 LGTM! Just will wait for an approval from another contributor.

/cc @javache @grabbou

File: React/Views/RCTTabBar.m [hunk 1]
```diff
@@ -105,6 +105,9 @@ - (void)reactBridgeDidFinishTransaction
     _tabController.viewControllers = viewControllers;
     _tabsChanged = NO;
   }
+  
+  [[UITabBarItem appearance] setTitleTextAttributes:[NSDictionary dictionaryWithObjectsAndKeys:_unselectedTintColor, NSForegroundColorAttributeName, nil] forState:UIControlStateNormal];
+  [[UITabBarItem appearance] setTitleTextAttributes:[NSDictionary dictionaryWithObjectsAndKeys:self.tintColor, NSForegroundColorAttributeName, nil] forState:UIControlStateSelected];
```
Comment (nicklockwood) [inline React/Views/RCTTabBar.m]: This is a global appearance attribute that will affect all tab bar items, right? Not just the ones in this tab bar.

I appreciate that it's unlikely that an app would have more than one tab bar, but I think we should find a better solution.

Ideally the RCTTabBar would apply these appearance attributes just to its own tab bar items.

**Comment by facebook-github-bot-3** [comment]
@urbancvek updated the pull request.

**Comment by urbancvek** [comment]
@nicklockwood you are right it would affect all TabBars. I've now applied it (#212b7c3) just a few lines below where it loops through children TabItems. It will apply styles only for a single TabBar now.

**Comment by facebook-github-bot-6** [comment]
@urbancvek updated the pull request.

**Comment by facebook-github-bot-3** [comment]
@urbancvek updated the pull request.

**Comment by grabbou** [comment]
Is it going to conflict with my PR about itemPositioning? One of them might need rebasing after we merge first one.

**Comment by urbancvek** [comment]
@grabbou There might be some conflicts but not in functionality. Will do a rebase if it's needed.

**Comment by qpowell** [comment]
@urbancvek great work!

**Comment by nicklockwood** [comment]
@facebook-github-bot shipit

**Comment by facebook-github-bot-8** [comment]
Thanks for importing. If you are an FB employee go to [Phabricator](https://our.intern.facebook.com/intern/opensource/github/pull_request/943756249079309/int_phab) to review.

**Comment by facebook-github-bot-3** [comment]
@urbancvek updated the pull request.

**Comment by facebook-github-bot-6** [comment]
@urbancvek updated the pull request.

**Comment by urbancvek** [comment]
@nicklockwood Hey. I tested a bit more what I have written and found out that it failes if unselectedTintColor prop is not passed in. (Figured that out in Navigator > Jumping Example). I added a check for this before it tries to add color to items here (77a1dd1). Is this why Import failed?

**Comment by nicklockwood** [comment]
@facebook-github-bot shipit

**Comment by nicklockwood** [comment]
@urbancvek I think so - import failed because a unit test crashed after it tried to init a dictionary with a nil element.

**Comment by mkonicek** [comment]
> I tested a bit more what I have written and found out that it failes if unselectedTintColor prop is not passed in.

Removed the Import Failed label because this was a legit error. For tracking purposes I'm using the Import Failed label for failures caused by flakiness only.

**Comment by cauchywei** [comment]
Hi @urbancvek ,I found the following code will remove unselectedIcon's gray tint color only, it can't affect selectedIcon so that the selectedIcon will always tinted. You can see my issue #7467 

```Objective-C
self.barItem.image = [_icon imageWithRenderingMode:UIImageRenderingModeAlwaysOriginal];
````

**Comment by urbancvek** [comment]
Hey @cauchywei! 😊

What my code does is that if you set renderAsOriginal prop to true it will take the icon it's given and render it. The same will happen with selectedIcon. This means you CANNOT control it with tintColor or unselectedTintColor. It will just ignore it. (it will be used for color of the text)

If you want to have the icon or selectedIcon the same color as your text (controlled by tintColor or unselectedTintColor) you just open photoshop or sketch or whatever you are using and make the icon look the way you want. Then just pass it in to TabBarIOS.Item. 

I don't know how you want to enhance the TabBarItem because it does everything you could possibly want.

**Comment by cauchywei** [comment]
@urbancvek let us about it in may issue #7467 😊

**Comment by mowi22** [comment]
Hi, Im a noob to react and your updates to the TabBar.I have two questions.

1. Im trying to use renderAsOriginal on the "selected" Tab item but having no success. (I don't want any tint on the selected icon). I only want the text to highlight and the icons to stay full color. renderAsOriginal is  working on the unselected items just fine. Thank you for that. 

2. I was hoping to just have the other unselected items dim in opacity. Is this possible trough this component or will i have to create dimmed versions in photoshop?

**Comment by urbancvek** [comment]
Hey @mowi22, sorry for the late answer. Didn't have much time over the weekend but I now implemented what you wanted. Previuosly the component only looked at icon and applied renderAsOriginal to it. It ignored the selectedIcon and rendered it as a template. I will create a new PR now so it will take a few days for it to get accepted.
Thank you for noticing this issue! :D

**Pull request closed without merging**